haystack/core/pipeline/base.py

```diff
@@ -308,6 +308,30 @@ def add_component(self, name: str, instance: Component) -> None:
             visits=0,
         )
 
+    def remove_component(self, name) -> None:
+        """
+        Remove a component from the pipeline.
+
+        Remove an existing component from the pipeline by providing its name.
+        All edges that connect to the component will also be deleted.
+
+        :param name:
+            The name of the component to remove.
+
+        :raises ValueError:
+            If there is no component with that name already in the Pipeline.
+        """
+
+        # Check that a component with that name is in the Pipeline
+        if name not in self.graph.nodes:
+            raise ValueError(
+                f"There is no component named '{name}' in the pipeline. The valid component names are: ",
+                ", ".join(n for n in self.graph.nodes),
+            )
```

Comment:
I wouldn't fail here. I think it's best if the function is idempotent, if the component is not there just return.

Reply:
But I believe that it is better to raise an error to allow automations to handle this accordingly.

Reply:
Can you give me an example?

Reply:
Imagine I am working on an agent that has to update a pipeline, generation might give a wrong component name, and then I would want to handle the error accordingly, for example by prompting the llm again.

Reply:
That could still be handled outside but it still a valid use case I guess.